bemani/utils/read.py

```diff
@@ -1620,73 +1648,250 @@ def scrape(self, binfile: str, assets_dir: Optional[str]) -> List[Dict[str, Any]
         finally:
             bh.close()
 
-        musicdb = IIDXMusicDB(binarydata)
+        import_qpros = True  # by default, try to import qpros
+        try:
+            pe = pefile.PE(data=binarydata, fast_load=True)
+        except BaseException:
+            import_qpros = False  # if it failed then we're reading a music db file, not the executable
+
+        def virtual_to_physical(offset: int) -> int:
```

Comment:
This really should be moved to some common functionality, like common.pefile or something. We've copy-pasted it so many times by now...

Reply:
https://github.com/DragonMinded/bemaniutils/issues/28 created for this, since its not applicable to this PR.